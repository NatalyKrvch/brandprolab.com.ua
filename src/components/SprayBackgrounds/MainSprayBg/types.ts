export interface MainSprayBgProps {
  className?: string;
}
