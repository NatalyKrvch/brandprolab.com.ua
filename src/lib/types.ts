export type Size = 's' | 'm' | 'l';
export type Screen = 'mobile' | 'tablet' | 'desktop';
