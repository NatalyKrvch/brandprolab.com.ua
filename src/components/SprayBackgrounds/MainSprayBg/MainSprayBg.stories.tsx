import type { Meta, StoryObj } from '@storybook/react';

import MainSprayBg from '.';

const meta: Meta<typeof MainSprayBg> = {
  title: 'Components/SprayBackground',
  component: MainSprayBg,
  tags: ['autodocs'],
  args: {
    className: 'top-[-100px] left-[-100px] w-[600px] h-[600px]',
  },
};

export default meta;

type Story = StoryObj<typeof MainSprayBg>;

export const Default: Story = {};
