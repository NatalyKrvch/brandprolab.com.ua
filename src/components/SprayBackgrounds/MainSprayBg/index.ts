export { default as MainSprayBg } from './MainSprayBg';
export * from './types';
