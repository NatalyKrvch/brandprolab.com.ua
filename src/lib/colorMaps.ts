import {
  MainSocialCardColor,
  socialMediaColorClassMapType,
} from '@/components/Cards/SocialMediaCard/types';

export const socialMediaColorClassMap: socialMediaColorClassMapType = {
  [MainSocialCardColor.RED]: {
    bg: 'bg-red-default',
    text: 'text-red-default',
  },
  [MainSocialCardColor.BLUE]: {
    bg: 'bg-blue-default',
    text: 'text-blue-default',
  },
};
