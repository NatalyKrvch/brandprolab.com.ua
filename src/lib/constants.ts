export const ALL_DATA_QUERY = `
{
  "header": *[_type == "header"][0],
  "hero": *[_type == "hero"][0],
  "cases": *[_type == "cases"][0],
  "services": *[_type == "services"][0],
  "testimonials": *[_type == "testimonialsSection"][0],
  "subscribe": *[_type == "subscribeSection"][0],
  "video": *[_type == "videoSection"][0],
  "expertise": *[_type == "expertiseSection"][0],
  "diagnostic": *[_type == "diagnosticSection"][0],
  "footer": *[_type == "footerSection"][0]
}
`;

export const YOUTUBE_REGEX =
  /(?:youtu\.be\/|youtube\.com\/(?:.*[?&]v=|embed\/|v\/|shorts\/|live\/))([^&?/]+)/;

export enum IconClassType {
  SERVICES = 'services',
  EXPERTISE = 'expertise',
  SOCIAL_MEDIA = 'social-media',
  SOCIAL_MEDIA_MAIN = 'social-media-main',
}
export const REVALIDATE = 60;
export const ENDLESS_CAROUSEL_SPEED = 50;
export const SLIDER_RIGHT_SHIFT = 2;
export const SLIDER_LEFT_SHIFT = -1;
export const AMOUNT_OF_TESTIMONIALS_WORDS = 15;
export const AMOUNT_OF_CASES_WORDS_DESKTOP = 13;
export const AMOUNT_OF_CASES_WORDS_TABLET = 9;
export const AMOUNT_OF_CASES_WORDS_MOBILE = 13;
