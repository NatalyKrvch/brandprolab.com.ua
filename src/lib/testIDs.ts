export const INTERNAL_LINK_TEST_ID = 'internal-link-test-id';
export const EXTERNAL_LINK_TEST_ID = 'external-link-test-id';
export const FULL_WIDTH_CONTAINER_TEST_ID = 'full-width-container-test-id';
export const LIMITED_WIDTH_CONTAINER_TEST_ID =
  'limited-width-container-test-id';
export const LEFT_LIMITED_CONTAINER_TEST_ID = 'left-limited-container-test-id';
export const ICON_COMPONENT_TEST_ID = 'icon-component-test-id';
export const ICON_COMPONENT_IMAGE_TEST_ID = 'icon-component-image-test-id';
export const LABEL_COMPONENT_TEST_ID = 'label-component-test-id';
export const BUTTON_TEST_ID = 'button-test-id';
export const UNDERLINED_TEXT_TEST_ID = 'underlined-text-test-id';
export const USER_BADGE_TEST_ID = 'user-badge-test-id';
export const USER_BADGE_IMAGE_TEST_ID = 'user-badge-image-test-id';
export const USER_BADGE_NAME_TEST_ID = 'user-badge-name-test-id';
export const TESTIMONIAL_CARD_TEST_ID = 'testimonials-card-test-id';
export const TESTIMONIAL_TEXT_TEST_ID = 'testimonials-text-test-id';
export const LIST_TEST_ID = 'list-test-id';
export const LIST_ITEM_TEST_ID = 'list-item-test-id';
export const SIMPLE_CARD_TEST_ID = 'simple-card-test-id';
export const DIPLOMA_CARD_TEST_ID = 'diploma-card-test-id';
export const DIPLOMA_IMAGE_TEST_ID = 'diploma-image-test-id';
export const DIPLOMA_OVERLAY_TEST_ID = 'diploma-overlay-test-id';
export const PHOTO_CARD_IMAGE_TEST_ID = 'photo-card-image-test-id';
export const PHOTO_CARD_TEXT_TEST_ID = 'photo-card-text-test-id';
export const VIDEO_IFRAME_TEST_ID = 'video-iframe-test-id';
export const VIDEO_PLAY_BUTTON_TEST_ID = 'video-play-button-test-id';
export const VIDEO_TEST_ID = 'video-test-id';
export const VIDEO_THUMBNAIL_TEST_ID = 'video-thumbnail-test-id';
export const FLIP_CARD_BACK_TEST_ID = 'flip-card-back-test-id';
export const FLIP_CARD_FRONT_TEST_ID = 'flip-card-front-test-id';
export const FLIP_CARD_TEST_ID = 'flip-card-test-id';
export const SERVICE_CARD_FRONT_DESCRIPTION_TEST_ID =
  'service-card-front-description-test-id';
export const SERVICE_CARD_FRONT_HEADER_TEST_ID =
  'service-card-front-header-test-id';
export const SERVICE_CARD_BACK_HEADER_TEST_ID =
  'service-card-back-header-test-id';
export const SERVICE_CARD_BACK_LIST_ITEM_TEST_ID =
  'service-card-back-list-item-test-id';
export const SERVICE_CARD_BACK_LIST_TEST_ID = 'service-card-back-list-test-id';
export const SOCIAL_MEDIA_LABEL_TEST_ID = 'social-media-label-test-id';
export const SOCIAL_MEDIA_DESCRIPTION_TEST_ID =
  'social-media-description-test-id';
export const FOOTER_SECTION_TEST_ID = 'footer-section-test-id';
export const PAGINATION_DOT_TEST_ID = 'pagination-dot-test-id';
export const CAROUSEL_NAV_WRAPPER_TEST_ID = 'carousel-nav-wrapper-test-id';
export const CAROUSEL_NAV_NEXT_TEST_ID = 'carousel-nav-next-test-id';
export const CAROUSEL_NAV_PREV_TEST_ID = 'carousel-nav-prev-test-id';
export const ENDLESS_CAROUSEL_TEST_ID = 'endless-carousel-test-id';
export const ENDLESS_CAROUSEL_CONTENT_TEST_ID =
  'endless-carousel-content-test-id';
export const MAIN_SPRAY_BG_TEST_ID = 'main-spray-bg-test-id';
export const HERO_MAIN_CARD_ICON_TEST_ID = 'hero-main-card-icon-test-id';
export const HERO_TEXT_CONTENT_TEST_ID = 'hero-text-content-test-id';
export const HERO_TEXT_CONTENT_ICON_TEST_ID = 'hero-text-content-icon-test-id';
export const EXPERTISE_SECTION_TEST_ID = 'expertise-section-test-id';
export const SCROLL_UP_BUTTON_TEST_ID = 'scroll-up-button-test-id';
export const MODAL_WRAPPER_TEST_ID = 'modal-wrapper-test-id';
export const MODAL_CONTENT_TEST_ID = 'modal-content-test-id';
export const MODAL_CLOSE_BUTTON_TEST_ID = 'modal-close-button-test-id';
export const CASE_MODAL_TEXT_TEST_ID = 'case-modal-text-test-id';
export const TESTIMONIAL_MODAL_NAME_TEST_ID = 'testimonial-modal-name-test-id';
export const TESTIMONIAL_MODAL_POSITION_TEST_ID =
  'testimonial-modal-position-test-id';
export const TESTIMONIAL_MODAL_TEXT_TEST_ID = 'testimonial-modal-text-test-id';
export const TESTIMONIAL_MODAL_IMAGE_TEST_ID =
  'testimonial-modal-image-test-id';
