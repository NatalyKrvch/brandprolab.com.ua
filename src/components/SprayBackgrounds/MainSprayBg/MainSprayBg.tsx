import Image from 'next/image';
import { twMerge } from 'tailwind-merge';

import { MAIN_SPRAY_BG_TEST_ID } from '@/lib/testIDs';

import type { MainSprayBgProps } from './types';

const MainSprayBg = ({ className = '' }: MainSprayBgProps) => (
  <Image
    src="/images/backgrounds/spray.svg"
    alt=""
    width={100}
    height={200}
    data-testid={MAIN_SPRAY_BG_TEST_ID}
    aria-hidden="true"
    className={twMerge(
      'pointer-events-none absolute inset-0 z-[-1] select-none overflow-hidden object-cover blur-2xl',
      className,
    )}
  />
);

export default MainSprayBg;
