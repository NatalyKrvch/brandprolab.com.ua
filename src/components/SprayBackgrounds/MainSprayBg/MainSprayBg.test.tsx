import { render } from '@testing-library/react';

import { MAIN_SPRAY_BG_TEST_ID } from '@/lib/testIDs';

import MainSprayBg from './MainSprayBg';

describe('MainSprayBg', () => {
  it('renders an img with correct default attributes', () => {
    const { getByTestId, asFragment } = render(<MainSprayBg />);

    const img = getByTestId(MAIN_SPRAY_BG_TEST_ID);
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', '/images/backgrounds/spray.svg');
    expect(img).toHaveAttribute('alt', '');
    expect(img).toHaveAttribute('aria-hidden', 'true');
    expect(img).toHaveClass(
      'pointer-events-none',
      'absolute',
      'inset-0',
      'z-[-1]',
      'select-none',
      'object-cover',
      'blur-2xl',
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom className if provided', () => {
    const { getByTestId, asFragment } = render(
      <MainSprayBg className="left-[200px] top-[100px]" />,
    );

    const img = getByTestId(MAIN_SPRAY_BG_TEST_ID);
    expect(img).toHaveClass('top-[100px]', 'left-[200px]');

    expect(asFragment()).toMatchSnapshot();
  });
});
