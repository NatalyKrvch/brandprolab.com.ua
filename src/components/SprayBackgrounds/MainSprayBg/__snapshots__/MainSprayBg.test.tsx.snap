// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSprayBg applies custom className if provided 1`] = `
<DocumentFragment>
  <img
    alt=""
    aria-hidden="true"
    class="pointer-events-none absolute inset-0 z-[-1] select-none overflow-hidden object-cover blur-2xl left-[200px] top-[100px]"
    data-nimg="1"
    data-testid="main-spray-bg-test-id"
    decoding="async"
    height="200"
    loading="lazy"
    src="/images/backgrounds/spray.svg"
    style="color: transparent;"
    width="100"
  />
</DocumentFragment>
`;

exports[`MainSprayBg renders an img with correct default attributes 1`] = `
<DocumentFragment>
  <img
    alt=""
    aria-hidden="true"
    class="pointer-events-none absolute inset-0 z-[-1] select-none overflow-hidden object-cover blur-2xl"
    data-nimg="1"
    data-testid="main-spray-bg-test-id"
    decoding="async"
    height="200"
    loading="lazy"
    src="/images/backgrounds/spray.svg"
    style="color: transparent;"
    width="100"
  />
</DocumentFragment>
`;
